### PR TITLE
Update to use RavenPy v0.8.1 and adjust documentation

### PR DESCRIPTION
**Pull request by Zeitsperre**
## Summary of changes

- Updated from RavenPy v0.7.8 to v0.8.1
- Dropped support for Python3.7 (same as in RavenPy)
- Added `rioxarray` and `ipywidgets` to requirements (used explicitly in notebooks)
- Removed a duplicated notebook (`Extract_geographical_watershed_properties.ipynb`) that was not in the existing documentation.
- Set notebooks under docs/source/paper to be ignored.
- Set ReadTheDocs to use `mambaforge` for installing dependencies (quicker/less memory intensive)
- Set ReadTheDocs builds to "fail-on-warning"

- Updated GitHub Workflows:
  * Use latest Ubuntu images
  * Linting to use Python3.8
  * Staged a Python3.10 build for testing purposes

* Updated pre-commit hooks:
  * Added pygrep-hooks, NBQA-hooks, yamlint, blackdoc, check-manifest,
GitHub and RtD schema validation
  * Ran hooks over code base

### CODE REVIEW

**Comment by Zeitsperre** [comment]
Running into an issue due to settings in ReadTheDocs. Relevant information here: https://github.com/readthedocs/readthedocs.org/issues/9306

File: .readthedocs.yml [hunk 1]
```diff
@@ -10,7 +10,7 @@ sphinx:
   configuration: docs/source/conf.py
   # fail_on_warning might generate hard to fix error, in this case it can be
   # disabled but this also means those errors will fail silently, choose wisely.
-  fail_on_warning: false
+  fail_on_warning: true
```
Comment (tlvu) [inline .readthedocs.yml:13]: Love this !  How many things to fix?

Comment (tlvu) [reply .readthedocs.yml:13]: Add `-W` to this line in  Makefile https://github.com/Ouranosinc/raven/blob/d0d0a6d4ae652230a9d1ae18e5a95d0d6ab4a4be/docs/Makefile#L23 to also match this `fail_on_warning: true`.

Same as in my original commit https://github.com/Ouranosinc/raven/commit/fa979fff9c38163bd08ff27c973636b4efb664a5

**Review by tlvu — APPROVED** [review]
LGTM

**Comment by tlvu** [comment]
> Running into an issue due to settings in ReadTheDocs. Relevant information here: [readthedocs/readthedocs.org#9306](https://github.com/readthedocs/readthedocs.org/issues/9306)

Oh they previously activate this `conda update` for us to ensure we use latest `conda` to have speed and memory optimization in latest conda.

Latest conda now not fast or use too much memory?  Have to switch to mamba?

**Comment by tlvu** [comment]
> > Running into an issue due to settings in ReadTheDocs. Relevant information here: [readthedocs/readthedocs.org#9306](https://github.com/readthedocs/readthedocs.org/issues/9306)
> 
> Oh they previously activate this `conda update` for us to ensure we use latest `conda` to have speed and memory optimization in latest conda.
> 
> Latest conda now not fast or use too much memory? Have to switch to mamba?

Here is when that conda update is activated for Raven: https://github.com/readthedocs/readthedocs.org/issues/5847#issuecomment-519198831

File: .readthedocs.yml (hunk 1 above)
Comment (Zeitsperre) [reply .readthedocs.yml:13]: > How many things to fix?

The unreferenced notebooks and adding `ipywidgets`. Not much!

**Comment by Zeitsperre** [comment]
> Oh they previously activate this `conda update` for us to ensure we use latest `conda` to have speed and memory optimization in latest conda.

Ah! I forgot that was done on this repo/project. My bad!

> Latest conda now not fast or use too much memory? Have to switch to mamba?

I figured why not have more synonymous buildouts for all of our documentation. It's not absolutely necessary, but it can't hurt?

**Comment by tlvu** [comment]
> > Latest conda now not fast or use too much memory? Have to switch to mamba?
> 
> I figured why not have more synonymous buildouts for all of our documentation. It's not absolutely necessary, but it can't hurt?

Not hurt but I guess now you'll have to open an issue for RtD dev to disable that conda update flag for mamba to work.

**Review by huard — APPROVED** [review]

File: raven/processes/wps_raven.py [hunk 2]
```diff
@@ -93,6 +93,7 @@ def options(self, request):
                 else:
                     data = obj.data
 
+                # FIXME: Breaking changes in RavenPy occurred at https://github.com/CSHS-CWRA/RavenPy/commit/8e21efb8e2f0e16a3e1a5fedeec098a4fc3c5d98
```
Comment (Zeitsperre) [inline raven/processes/wps_raven.py:96]: @huard

I'm not sure how best to proceed here. This change is causing the majority of errors for the time being. I won't pretend to understand how to proceed on this.

**Pull request merged**